src/int/sign.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{ConstChoice, ConstCtOption, Int, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Construct new [`Int`] from a sign and magnitude.
+    /// Returns `None` when the magnitude does not fit in an [`Int<LIMBS>`].
+    pub const fn new_from_abs_sign(
+        abs: Uint<LIMBS>,
+        is_negative: ConstChoice,
+    ) -> ConstCtOption<Self> {
+        let magnitude = Self(abs).wrapping_neg_if(is_negative);
+        let fits = Uint::gt(&abs, &Int::MAX.0)
+            .not()
+            .or(is_negative.and(Uint::eq(&abs, &Int::MIN.0)));
+        ConstCtOption::new(magnitude, fits)
+    }
+
+    /// Whether this [`Int`] is negative, as a `ConstChoice`.
+    pub const fn is_negative(&self) -> ConstChoice {
+        ConstChoice::from_word_msb(self.0.to_words()[LIMBS - 1])
+    }
```

Comment:
I know you copied this from my code, but perhaps we should use something like `.get(LIMBS-1).unwrap_or(ZERO)` to deal with the (illegal) `LIMBS=0` case? wdyt?

Reply:
Int will always have at least one limb, so there always exist most significant limb.
Maybe we can define a method to return this most significant limb?

Reply:
maybe compile time assertion `LIMBS > 0`?
I know mathematically Int<0> should exist and it is the type with only one element (namely `0`) but I don't think we want to support this, correct me if I'm wrong.

Reply:
Using something like [static assertions](https://docs.rs/static_assertions/latest/static_assertions/) indeed sounds like a better solution. Could you please add these?

Reply:
@erik-3milabs I decided not to use static_assertions to avoid unnecessary dependencies, I did however done this semantically correct (i.e. by returning zero when there are no limbs).